examples/kcore/test_kcore.cu

```diff
@@ -0,0 +1,142 @@
+// ----------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------
+
+/**
+ * @file
+ * test_kcore.cu
+ *
+ * @brief Simple test driver program for vertex k-core decomposition.
+ */
+
+#include <gunrock/app/kcore/kcore_app.cu>
+#include <gunrock/app/test_base.cuh>
+
+using namespace gunrock;
+
+namespace APP_NAMESPACE = app::kcore;
+
+/******************************************************************************
+ * Main
+ ******************************************************************************/
+
+/**
+ * @brief Enclosure to the main function
+ */
+struct main_struct {
+  /**
+   * @brief the actual main function, after type switching
+   * @tparam VertexT    Type of vertex identifier
+   * @tparam SizeT      Type of graph size, i.e. type of edge identifier
+   * @tparam ValueT     Type of edge values
+   * @param  parameters Command line parameters
+   * @param  v,s,val    Place holders for type deduction
+   * \return cudaError_t error message(s), if any
+   */
+  template <typename VertexT,  // Use int as the vertex identifier
+            typename SizeT,    // Use int as the graph size type
+            typename ValueT>   // Use int as the value type
+  cudaError_t
+  operator()(util::Parameters &parameters, VertexT v, SizeT s, ValueT val) {
+    // CLI parameters
+    bool quick = parameters.Get<bool>("quick");
+    bool quiet = parameters.Get<bool>("quiet");
+    parameters.Set("undirected", true);
```

Comment:
So you require graphs to be undirected?

Reply:
Yes, mainly because that's what I need for my maximum clique implementation. Lol. It's also what I've seen in the few other k-core papers I've looked at. But I think an "in-degree k-core" or "out-degree k-core" are concepts that exist, and it would be nice to add versions that compute those in the future.

Reply:
Confirmed that directional k-core decomposition does exist: https://rdrr.io/cran/igraph/man/coreness.html